spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/messaging/MessagingSpanExtractor.java

```diff
@@ -100,7 +91,7 @@ private void setParentIdIfApplicable(Message<?> carrier, SpanBuilder spanBuilder
 				spanBuilder.parent(Span.hexToId(parentId));
 			}
 		} catch (Exception e) {
-			log.warn("Exception occurred while trying to set parentId", e);
+			throw new IllegalStateException("Malformed id", e);
```

Comment:
prefer IllegalArgumentException as IllegalStateException is more temporal in nature (ex a state issue that could resolve itself)

the catch Exception is a bit wide, too.. one way could be to make the Span.hexToId throw IllegalArgumentException, and propagate all IllegalArgumentExceptions instead of wrapping them